src/net/sourceforge/kolmafia/session/GreyYouManager.java

```diff
@@ -124,6 +124,32 @@ public static GooSkill learnSkill(int skillId) {
     return skill;
   }
 
+  public static void absorbMonster(MonsterData monster, String absorbText) {
+    // This called from FightRequest after a win
+    if (monster == null || !KoLCharacter.inGreyYou()) {
+      return;
+    }
+
+    Absorption absorb = allAbsorptions.get(monster.getId());
+
+    if (absorb == null) {
+      return;
+    }
+
+    // If absorb is expected to give adventures, but neither message for adventure absorbs is found
+    if (absorb.getType() == AbsorptionType.ADVENTURES
```

Comment:
might be slightly cleaner to split this one out into its own `if()` then indent the rest under it, so the check only gets called once and combines with the preference increase below